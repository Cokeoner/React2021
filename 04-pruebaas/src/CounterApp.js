import React, { useState } from 'react'

import PropTypes from 'prop-types'

const CounterApp = ( { value = 10 } ) => {

    const [counter, setCounter ] = useState( value );
  
    // function setCounter() => {

    // };


    //handeAdd
    const handeAdd = (e) => { 
        setCounter(counter + 1);
        // setCounter((c) => c + 1);
    } 

    const handeReset = (e) => setCounter(value); 


    return(
            <div>
                <h1>Counter</h1>
                <h2> { counter } </h2>
                <button onClick={ handeAdd }> +1 </button>
                <button onClick={ handeReset }> Reset </button>
                <button onClick={ (e) =>{ setCounter(counter - 1)} }> -1 </button>
            </div>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number
}


export default CounterApp;


 