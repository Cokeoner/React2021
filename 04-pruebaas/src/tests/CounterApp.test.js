import React from 'react';
import { shallow } from 'enzyme'
// import { render } from "@testing-library/react";
import '@testing-library/jest-dom';
import CounterApp from  "../CounterApp";

describe('Debe de mostrar el CounterApp', () => {

    test('Valores por Defecto', () => {
        // const saludo = 'Hola soy goku';
        const Wrapper = shallow( <CounterApp />)

        expect( Wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar el valor 100', () => {
        
        const valors = '100';

        const wrapper = shallow(
                <CounterApp 
                  value = { valors }
                />);
        const conteeValors = wrapper.find('h2').text().trim();

        expect(conteeValors).toBe(valors);

    })
    
});
