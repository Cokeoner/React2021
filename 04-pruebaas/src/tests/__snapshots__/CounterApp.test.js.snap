// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de mostrar el CounterApp Valores por Defecto 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    onClick={[Function]}
  >
     -1 
  </button>
</div>
`;
